Cannot build examples using the module "sock_tcp"

#### Description
Cannot build examples using the module "sock_tcp"

#### Steps to reproduce the issue
Build any example using the module sock_tcp

#### Expected results
The example builds without errors.


#### Actual results
We get this error:

```
/home/user/riot/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c: In function 'gnrc_tcp_ep_from_str':
/home/user/riot/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c:187:24: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  187 |     char *addr_begin = strchr(str, '[');
      |                        ^~~~~~
/home/user/riot/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c:188:22: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  188 |     char *addr_end = strchr(str, ']');
      |                      ^~~~~~
/home/user/riot/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c:232:22: error: initialization discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  232 |     char *if_begin = strchr(str, '%');
```


#### Versions

```
RIOT version information
----------------------------
              RIOT branch: master
         RIOT commit hash: 95823dc3e8b13c894770e5d0a3d38ea38dcaadbc
         RIOT commit date: 2026-04-23

Operating System Environment
----------------------------
         Operating System: "Arch Linux" 
                   Kernel: Linux 6.19.14-arch1-1 x86_64 unknown
             System shell: GNU bash, Version 5.3.9(1)-release (x86_64-pc-linux-gnu)
             make's shell: GNU bash, Version 5.3.9(1)-release (x86_64-pc-linux-gnu)

Installed compiler toolchains
-----------------------------
               native gcc: gcc (GCC) 15.2.1 20260209
        arm-none-eabi-gcc: arm-none-eabi-gcc (Arch Repository) 14.2.0
                  avr-gcc: missing
           msp430-elf-gcc: missing
       riscv-none-elf-gcc: missing
  riscv64-unknown-elf-gcc: missing
      riscv32-esp-elf-gcc: missing
  riscv32-unknown-elf-gcc: missing
     xtensa-esp32-elf-gcc: missing
   xtensa-esp32s2-elf-gcc: missing
   xtensa-esp32s3-elf-gcc: missing
   xtensa-esp8266-elf-gcc: missing
                    clang: clang version 22.1.3

Installed compiler libs
-----------------------
     arm-none-eabi-newlib: "4.5.0"
        msp430-elf-newlib: missing
    riscv-none-elf-newlib: missing
riscv64-unknown-elf-newlib: missing
   riscv32-esp-elf-newlib: missing
riscv32-unknown-elf-newlib: missing
  xtensa-esp32-elf-newlib: missing
xtensa-esp32s2-elf-newlib: missing
xtensa-esp32s3-elf-newlib: missing
xtensa-esp8266-elf-newlib: missing
                 avr-libc: missing (missing)

Installed development tools
---------------------------
                   ccache: ccache version 4.13.4
                    cmake: cmake version 4.3.2
                 cppcheck: Cppcheck 2.20.0
                  doxygen: missing
                      git: git version 2.54.0
                     make: GNU Make 4.4.1
                  openocd: Open On-Chip Debugger 0.12.0-01004-g9ea7f3d64-dirty (2025-11-12-08:18)
                   python: Python 3.14.4
                  python2: missing
                  python3: Python 3.14.4
                   rustup: missing
                    cargo: missing
                    rustc: missing
                   c2rust: missing
                     node: v22.22.2
                      npm: 11.13.0
                   flake8: error: /usr/sbin/python3: No module named flake8
               coccinelle: missing
```

--- a/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c
+++ b/sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c
@@ -184,8 +184,8 @@ int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
     unsigned netif = 0;
 
     /* Examine given string */
-    char *addr_begin = strchr(str, '[');
-    char *addr_end = strchr(str, ']');
+    const char *addr_begin = strchr(str, '[');
+    const char *addr_end = strchr(str, ']');
 
     /* 1) Ensure that str contains a single pair of brackets */
     if (!addr_begin || !addr_end || strchr(addr_begin + 1, '[') || strchr(addr_end + 1, ']')) {
@@ -201,7 +201,7 @@ int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
     }
 
     /* 3) Examine optional port number */
-    char *port_begin = strchr(addr_end, ':');
+    const char *port_begin = strchr(addr_end, ':');
     if (port_begin) {
         /* 3.1) Ensure that there are characters left to parse after ':'. */
         if (*(++port_begin) == '\0') {
@@ -211,7 +211,7 @@ int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
         }
 
         /* 3.2) Ensure that port is a number (atol, does not report errors) */
-        for (char *ptr = port_begin; *ptr; ++ptr) {
+        for (const char *ptr = port_begin; *ptr; ++ptr) {
             if ((*ptr < '0') || ('9' < *ptr)) {
                 TCP_DEBUG_ERROR("-EINVAL: Invalid address string.");
                 TCP_DEBUG_LEAVE;
@@ -229,7 +229,7 @@ int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
     }
 
     /* 4) Examine optional interface identifier. */
-    char *if_begin = strchr(str, '%');
+    const char *if_begin = strchr(str, '%');
     if (if_begin) {
         /* 4.1) Ensure that the identifier is not empty and within brackets. */
         if (addr_end <= (++if_begin)) {
@@ -239,7 +239,7 @@ int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
         }
 
         /* 4.2) Ensure that the identifier is a number (atol, does not report errors) */
-        for (char *ptr = if_begin; ptr != addr_end; ++ptr) {
+        for (const char *ptr = if_begin; ptr != addr_end; ++ptr) {
             if ((*ptr < '0') || ('9' < *ptr)) {
                 TCP_DEBUG_ERROR("-EINVAL: Invalid address string.");
                 TCP_DEBUG_LEAVE;

